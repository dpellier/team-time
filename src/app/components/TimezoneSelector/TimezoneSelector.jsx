
import React from 'react';
import PropTypes from 'prop-types';
import { getTimezones } from '../../utils/time';

const TimezoneSelector = ({ currentValue, id, onChange }) =>
    <select className='timezone-selector'
            id={ id }
            onChange={ onChange }
            defaultValue={ currentValue }>
        {
            getTimezones().map((timezone, idx) =>
                <option key={ idx }
                        value={ timezone }>
                    { timezone }
                </option>
            )
        }
    </select>
;

TimezoneSelector.propTypes = {
    currentValue: PropTypes.string,
    id: PropTypes.string,
    onChange: PropTypes.func
};

export { TimezoneSelector };
