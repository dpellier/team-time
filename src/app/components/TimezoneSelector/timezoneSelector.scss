
@import '../../style/form';

.timezone-selector {
    @include select();
}
