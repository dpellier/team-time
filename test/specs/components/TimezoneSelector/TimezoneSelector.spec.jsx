
const mockTimezones = [1, 2, 3, 4, 5];
jest.mock('../../../../src/app/utils/time', () => ({
    getTimezones: () => mockTimezones
}));

import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { TimezoneSelector } from '../../../../src/app/components/TimezoneSelector/TimezoneSelector';

configure({ adapter: new Adapter() });

describe('<TimezoneSelector />', () => {
    const mockValue = '12:00';
    const mockOnChange = jest.fn();

    it('is correctly rendered', () => {
        const component = renderer.create(
            <TimezoneSelector currentValue={ mockValue }
                              onChange={ mockOnChange } />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should display an option for every timezone', () => {
        const wrapper = shallow(
            <TimezoneSelector currentValue={ mockValue }
                              onChange={ mockOnChange } />
        );
        expect(wrapper.find('option').length).toBe(mockTimezones.length);
    });
});
