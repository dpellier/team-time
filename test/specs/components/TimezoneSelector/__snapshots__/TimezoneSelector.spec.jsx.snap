// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimezoneSelector /> is correctly rendered 1`] = `
<select
  className="timezone-selector"
  defaultValue="12:00"
  onChange={[MockFunction]}
>
  <option
    value={1}
  >
    1
  </option>
  <option
    value={2}
  >
    2
  </option>
  <option
    value={3}
  >
    3
  </option>
  <option
    value={4}
  >
    4
  </option>
  <option
    value={5}
  >
    5
  </option>
</select>
`;
